### ISSUE TITLE
ERROR: "too many colons in address" after updating to docker 20.10.6

### ISSUE THREAD

**Issue by abcalphabet** [docker/compose#8282]
## Description of the issue

After upgrading to docker 20.10.6 I get the following error when starting containers with docker-compose (`docker-compose up -d`):

```
ERROR: for toxiproxy  Cannot start service toxiproxy: driver failed programming external connectivity on endpoint my_project_toxiproxy_1 (...): Error starting userland proxy: error while calling PortManager.AddPort(): listen tcp: address :::9315: too many colons in address
```

The above error occurs for all containers (port numbers differ), running the containers via `docker run` works normally though.

This issue does not occur anymore when downgrading to docker 20.10.5. 

Docker is run as `rootless-docker`.

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.29.0, build unknown
docker-py version: 5.0.0
CPython version: 3.9.2
OpenSSL version: OpenSSL 1.1.1k  25 Mar 2021
```

**Output of `docker version`**
```
Client:
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.16.3
 Git commit:        370c28948e
 Built:             Mon Apr 12 14:10:41 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.3
  Git commit:       8728dd246c
  Built:            Mon Apr 12 14:10:25 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e.m
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
services:
  my_project:
    build:
      args:
        CI_GIT_CREDENTIALS:'DELETED'
      context: /home/user/go/src/github.com/asdf/my_project
      dockerfile: Dockerfile.dev
    command: make dev-run
    depends_on:
      node-exporter:
        condition: service_started
      prometheus:
        condition: service_started
    environment:
      ENV_CONFIG: IRRELEVANT
    ports:
    - published: 4205
      target: 4205
    restart: always
  node-exporter:
    image: prom/node-exporter
    ports:
    - published: 9100
      target: 9100
  prometheus:
    image: prom/prometheus
    ports:
    - published: 9090
      target: 9090
    volumes:
    - /home/user/go/src/github.com/asdf/my_project/prometheus.yaml:/etc/prometheus/prometheus.yml:rw
version: '3.8'
```
The config should not matter, as the issue occurs with all projects I've tried, so I chose the shortest one to paste here

## Steps to reproduce the issue

1. `dc up -d prometheus` with above config and software versions

### Observed result

Container does not start successfully.

### Expected result

Container starts successfully.

### Stacktrace / full error message

```
ERROR: for my_project_prometheus_1  Cannot start service prometheus: driver failed programming external connectivity on endpoint my_project_prometheus_1 (3c81c90f135ab413603020c4003d78d33440b0ee48539474dfa4d3176b30b523): Error starting userland proxy: error while calling PortManager.AddPort(): listen tcp: address :::9090: too many colons in address

ERROR: for prometheus  Cannot start service prometheus: driver failed programming external connectivity on endpoint my_project_prometheus_1 (3c81c90f135ab413603020c4003d78d33440b0ee48539474dfa4d3176b30b523): Error starting userland proxy: error while calling PortManager.AddPort(): listen tcp: address :::9090: too many colons in address
ERROR: Encountered errors while bringing up the project.
```

## Additional information

```
 Kernel Version: 5.11.13-arch1-1
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
```

**Comment by thaJeztah** [comment]
Can you check what version of rootlesskit you have installed? I think this was fixed in https://github.com/rootless-containers/rootlesskit/pull/213 (which is in rootlesskit v0.13.1 and up). Current version of rootlesskit in our `docker-ce-rootless-extras` package should be v0.14.1;


```
rootlesskit --version
rootlesskit version 0.14.1
```

**Comment by abcalphabet** [comment]
I was on version `0.11.0`, so it's looking like that's the most likely issue. [`docker-rootless-extras-bin` was updated yesterday evening](https://aur.archlinux.org/cgit/aur.git/log/?h=docker-rootless-extras-bin), will report back whether that fixes the issue.

**Comment by abcalphabet** [comment]
It does not, what they did do though is add a new dependency check that now leads to the following error when trying to upgrade:
```
installing docker (1:20.10.6-1) breaks dependency 'docker<1:20.10.6' required by docker-rootless-extras-bin
``` 

You said this issue should be fixed with fixed with rootlesskit version `0.13.1`, right? Yesterday evening's update upgraded to that version, so theoretically it should fix the issue I was having after upgrading to `20.10.6-1`, shouldn't it? Will open a ticket with Arch if this is the case.

Slightly related question, is [running rootless mode via `docker-rootless-extras-bin`](https://wiki.archlinux.org/index.php/docker#Docker_rootless) still the preferred method of running it on Arch? Otherwise I'll mark the page out of date as well.

**Comment by abcalphabet** [comment]
Regardless of remaining problems, this is not a `docker-compose` issue.
